accepted/0000-workspaces.md

```diff
@@ -0,0 +1,479 @@
+
+
+```
+$ tree
+.
+├── apps
+│  ├── x
+│  │  └── node_modules
+│  │     ├── bar -> ../../../node_modules/bar
+│  │     ├── baz -> ../../../node_modules/baz
+│  │     └── foo -> ../../../node_modules/foo
+│  ├── y
+│  │  └── node_modules
+│  │     ├── bar -> ../../../node_modules/bar
+│  │     ├── baz -> ../../../node_modules/baz
+│  │     └── foo -> ../../../node_modules/foo
+│  └── z
+│     └── node_modules
+│        ├── bar -> ../../../node_modules/bar
+│        ├── baz -> ../../../node_modules/baz
+│        └── foo -> ../../../node_modules/foo
+├── node_modules
+│  ├── bar -> ../packages/bar
+│  ├── baz -> ../packages/baz
+│  ├── foo -> ../packages/foo
+│  ├── x -> ../apps/x
+│  ├── y -> ../apps/y
+│  └── z -> ../apps/z
+├── package.json
+├── packages
+│  ├── bar
+│  │  └── node_modules
+│  │     ├── baz -> ../../../node_modules/baz
+│  │     └── foo -> ../../../node_modules/foo
+│  ├── baz
+│  │  └── node_modules
+│  │     ├── bar -> ../../../node_modules/bar
+│  │     └── foo -> ../../../node_modules/foo
+│  └── foo
+│     └── node_modules
+│        ├── bar -> ../../../node_modules/bar
+│        └── baz -> ../../../node_modules/baz
+├── plugins
+│  ├── a
+│  │  └── node_modules
+│  │     ├── baz -> ../../node_modules/baz
+│  │     └── foo -> ../../node_modules/foo
+│  ├── b
+│  │  └── node_modules
+│  │     ├── baz -> ../../node_modules/baz
+│  │     └── foo -> ../../node_modules/foo
+│  ├── c
+│  │  └── node_modules
+│  │     ├── baz -> ../../node_modules/baz
+│  │     └── foo -> ../../node_modules/foo
+│  ├── d
+│  │  └── node_modules
+│  │     ├── baz -> ../../node_modules/baz
+│  │     └── foo -> ../../node_modules/foo
+│  ├── node_modules
+│  │  ├── baz -> ../../node_modules/baz
+│  │  └── foo -> ../../node_modules/foo
+│  └── package.json
+└── ws2
+   ├── apps
+   │  └── twoapp
+   │     └── node_modules
+   │        └── two -> ../../../node_modules/two
+   ├── node_modules
+   │  ├── bar -> ../../node_modules/bar
+   │  ├── big-external-dep
+   │  ├── two -> ../packages/two
+   │  └── twoapp -> ../apps/twoapp
+   ├── package.json
+   └── packages
+      └── two
+         └── node_modules
+            ├── bar -> ../../../node_modules/bar
+            └── big-external-dep -> ../../../node_modules/big-external-dep
+```
+
+
+## Prior Art
+
+- [Lerna](https://github.com/lerna/lerna)
+- [Yarn workspaces](https://classic.yarnpkg.com/en/docs/workspaces/)
+- [Pnpm workspaces](https://pnpm.js.org/en/workspaces)
+
+## Unresolved Questions and Bikeshedding
+
+- Should we add a Workspace class (subclass of Node) in Arborist?
+- For this initial implementation there's no intention of adding a more ellaborate version/publish subcommand/workflow that would allow for bumping versions of nested packages + updating dependency references elsewhere across a workspace (similar to `lerna --independent` publish workflow).
```

Comment:
```suggestion
- For this initial implementation there's no intention of adding a more ellaborate version/publish subcommand/workflow that would allow for bumping versions of nested packages + updating dependency references elsewhere across a workspace (similar to `lerna --independent` publish workflow).
- Support for non-nested file structure workspaces (ex: adjacent directories)
```

I have a bunch of opinions here to share and want to take the time to fully write them up.  But I wanted to add this to the questions in case other folks had ideas here to share.

Reply:
Nota bene: `--independent` has nothing to do whatsoever with Lerna's cross-dependency change detection, it just lets you choose the version bump per-package instead of per-monorepo.

And yes, npm's initial implementation of workspaces should stay far, _far_ away from this certifiable bug farm.

Reply:
This change might be missleading.  You cannot "add a line" without "editing the line before or after.  My recommendation was just to add the open question:

```
- Support for non-nested file structure workspaces (ex: adjacent directories)
```

Reply:
I'll go ahead and accept this item since it's mostly an "Unresolved question" still